File: src/upper/ascii/ascii.py
import os

def version():
    try:
        with open('.version', 'r') as file:
            lines = file.readlines()
            for line in lines:
                if line.startswith('VERSION='):
                    return line.strip()[8:]
    except FileNotFoundError:
        print("Error: .version file not found.")
    except Exception as e:
        print(f"Error reading version: {e}")
    
    return "Unknown"

def get_username():
    return os.getenv('USER') or os.getenv('LOGNAME') or os.getenv('USERNAME') or 'Unknown'

def RetrieveDistributionName(Distro, DistroSize) -> None:
    try:
        with open("/etc/os-release", "r") as fp:
            for line in fp:
                if line.startswith("NAME="):
                    nameStart = line.find('=')
                    if nameStart != -1:
                        nameStart += 1 
                        if line[nameStart] == '"':
                            nameStart += 1
                            nameEnd = line.find('"', nameStart)
                            if nameEnd != -1:
                                Distro[:nameEnd - nameStart] = line[nameStart:nameEnd]
                                return
        Distro[:len("Unknown")] = "Unknown"
    except Exception:
        Distro[:len("Unknown")] = "Unknown"

def ascii():
    Username = get_username()

    distro = ["" for _ in range(256)]
    RetrieveDistributionName(distro, len(distro))
    distro = ''.join(distro).strip()

    print("⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⣤⡀⠀⠀⠀⠀⠀")
    print("⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢻⣧⠃⡃⠀⠀⠀⠀⠀")
    print("⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢸⢿⠇⠏⡇⠀⠀⠀⠀")
    print("⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢠⣟⠏⡜⠰⢣⠀⠀⠀⠀")
    print("⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢀⡴⠟⣡⠊⡠⢁⣎⠀⠀⠀⠀")
    print("⠀⠀⢀⣶⣰⡂⠀⠀⠀⠀⣠⠖⠋⡔⠊⡠⢏⡠⢃⡜⠀⠀⠀⠀")
    print("⠀⠀⡧⣩⠾⢹⣓⣤⠀⠘⣯⡀⠀⣗⣋⣤⢃⠔⢩⠆⠀⠀⠀⠀")
    print("⠀⣸⣅⠁⢁⡞⠨⡍⢷⢂⡾⠁⣰⠡⠤⣊⠥⠒⠁⠀⠀⠀⠀⠀")
    print("⠘⠿⠽⠞⢫⠀⠀⠀⢻⠟⣡⡾⠱⡑⠦⠥⣀⣀⣀⠀⠀⠀⠀⠀")
    print("⠀⠀⠀⠀⡎⢀⠐⠀⠁⠈⠺⠷⠣⠃⠁⡀⠈⢿⡇⠨⢢⠀⠀⠀")
    print("⠀⠀⠀⢀⡃⠀⠐⠀⠂⠐⠂⢀⠂⠂⠐⢀⠀⡾⢔⢄⠡⣃⠀⠀")
    print(f"⠀⢸⡏⠃⠀⠙⣞⡆⠀⠀⠀⠀⠀⠀⣵⡚⠱⣎⣇⠀⠀   username: {Username}")
    print(f"⠀⣿⣷⠄⠀⠀⢹⣽⠀⠀⠀⠀⠀⣼⣳⠃⠀⠹⣾⣤⠀   distro: {distro}")
    print(f"⠀⠉⠁⠀⠀⠰⠿⠟⠀⠀⠀⠀⠘⠛⠛⠀⠀⠀⠹⠋    version: {version()}")

File: src/util/ascii.go
package util

import (
	"fmt"
	"os"
	"strings"
)

var (
	distro string
)

func version() string {
	data, err := os.ReadFile(".version")
	if err != nil {
		fmt.Println("Error: .version file not found.")
		return "Unknown"
	}
	lines := strings.Split(string(data), "\n")
	for _, line := range lines {
		if strings.HasPrefix(line, "VERSION=") {
			return strings.TrimPrefix(line, "VERSION=")
		}
	}
	fmt.Println("Error: Version not found in .version file.")
	return "Unknown"
}

func getUsername() string {
	username := os.Getenv("USER")
	if username == "" {
		username = os.Getenv("LOGNAME")
	}
	if username == "" {
		username = os.Getenv("USERNAME")
	}
	if username == "" {
		username = "Unknown"
	}
	return username
}

func retrieveDistributionName(distro *string) {
	data, err := os.ReadFile("/etc/os-release")
	if err != nil {
		*distro = "Unknown"
		return
	}
	lines := strings.Split(string(data), "\n")
	for _, line := range lines {
		if strings.HasPrefix(line, "NAME=") {
			nameStart := strings.Index(line, "=")
			if nameStart != -1 {
				nameStart++
				if line[nameStart] == '"' {
					nameStart++
					nameEnd := strings.Index(line[nameStart:], "\"")
					if nameEnd != -1 {
						*distro = line[nameStart : nameStart+nameEnd]
						return
					}
				}
			}
		}
	}
	*distro = "Unknown"
}

func Ascii() {
	username := getUsername()
	retrieveDistributionName(&distro)

	fmt.Println("⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⣤⡀⠀⠀⠀⠀⠀")
	fmt.Println("⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢻⣧⠃⡃⠀⠀⠀⠀⠀")
	fmt.Println("⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢸⢿⠇⠏⡇⠀⠀⠀⠀")
	fmt.Println("⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢠⣟⠏⡜⠰⢣⠀⠀⠀⠀")
	fmt.Println("⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢀⡴⠟⣡⠊⡠⢁⣎⠀⠀⠀⠀")
	fmt.Println("⠀⠀⢀⣶⣰⡂⠀⠀⠀⠀⣠⠖⠋⡔⠊⡠⢏⡠⢃⡜⠀⠀⠀⠀")
	fmt.Println("⠀⠀⡧⣩⠾⢹⣓⣤⠀⠘⣯⡀⠀⣗⣋⣤⢃⠔⢩⠆⠀⠀⠀⠀")
	fmt.Println("⠀⣸⣅⠁⢁⡞⠨⡍⢷⢂⡾⠁⣰⠡⠤⣊⠥⠒⠁⠀⠀⠀⠀⠀")
	fmt.Println("⠘⠿⠽⠞⢫⠀⠀⠀⢻⠟⣡⡾⠱⡑⠦⠥⣀⣀⣀⠀⠀⠀⠀⠀")
	fmt.Println("⠀⠀⠀⠀⡎⢀⠐⠀⠁⠈⠺⠷⠣⠃⠁⡀⠈⢿⡇⠨⢢⠀⠀⠀")
	fmt.Println("⠀⠀⠀⢀⡃⠀⠐⠀⠂⠐⠂⢀⠂⠂⠐⢀⠀⡾⢔⢄⠡⣃⠀⠀")
	fmt.Printf("⠀⢸⡏⠃⠀⠙⣞⡆⠀⠀⠀⠀⠀⠀⣵⡚⠱⣎⣇⠀⠀   username: %s\n", username)
	fmt.Printf("⠀⣿⣷⠄⠀⠀⢹⣽⠀⠀⠀⠀⠀⣼⣳⠃⠀⠹⣾⣤⠀   distro: %s\n", distro)
	fmt.Printf("⠀⠉⠁⠀⠀⠰⠿⠟⠀⠀⠀⠀⠘⠛⠛⠀⠀⠀⠹⠋    version: %s\n", version())
}
